### ISSUE TITLE
Log file not found when building from unsaved file

### ISSUE THREAD

**Issue by caspre24** [SublimeText/LaTeXTools#795]
When I invoke the LaTeX build via `CTRL+B` on an unsaved file, the log file cannot be found:

```
TraditionalBuilder: Engine: pdflatex. Invoking latexmk... done.

Could not find log file myTexFile.log!
Output from compilation:
```
The output of the latexmk log follows.
Nevertheless, the PDF is correctly generated and the latexmk log shows no errors.

If I save the .tex file before I start the build, everthing works fine.

**Comment by r-stein** [comment]
Do you use ST2 oder ST3?

**Comment by caspre24** [comment]
I'm using ST3 build 3114 on Ubuntu 16.04

**Comment by r-stein** [comment]
This is weird, because the first thing we do in the build is saving the file, if it is dirty.

**Comment by caspre24** [comment]
That was my thought too. And indeed, the dirty editor gets saved when starting the build, thus the PDF is correctly generated.

**Comment by ig0774** [comment]
The `latexmk` dump should have a line that says `Output written to myTeXFile.log`. Is that there? The other thing to check is if it happens if you delete the temporary files and then execute the build with an unsaved file. Maybe something is happening in the `aux` file that changes things?

**Comment by caspre24** [comment]
The error occurs the same way with temporary files deleted. I tried different combinations of temp files and/or pdf deleted and saved/unsaved editor. The saved/unsaved state seems the only relevant factor.

This is the last part of the latexmk dump
```
...
Output written on /home/carsten/git/rtsys/theses/csp-mt/thesis/csp-mt.pdf (34 pages, 403275 bytes).
SyncTeX written on /home/carsten/git/rtsys/theses/csp-mt/thesis/csp-mt.synctex.gz.
Transcript written on /home/carsten/git/rtsys/theses/csp-mt/thesis/csp-mt.log.
'csp-mt.fls' and '/home/carsten/git/rtsys/theses/csp-mt/thesis/csp-mt.fls' are identical (not copied) at /usr/bin/latexmk line 6657.
Latexmk: Found input bbl file 'csp-mt.bbl'
Latexmk: Log file says output to 'csp-mt.pdf'
Latexmk: ===For rule 'pdflatex', actual output 'csp-mt.pdf'
       ======appears not to match expected output '/home/carsten/git/rtsys/theses/csp-mt/thesis/csp-mt.pdf'
Latexmk: Found biber source file(s) [../csp-mt.bib csp-mt.bcf]
Latexmk: Undoing directory change
Latexmk: All targets (/home/carsten/git/rtsys/theses/csp-mt/thesis/csp-mt.pdf) are up-to-date
```
The full log can be found here http://pastebin.com/1MyCChsX

I tested a minimal document
```
\documentclass{book}
\begin{document}
  asdf
\end{document}
```
but even then, the same error persists when building from an unsaved editor.
Here is the corresponding log:
```
[Compiling /home/carsten/git/rtsys/theses/csp-mt/thesis/test.tex]

TraditionalBuilder: Engine: pdflatex. Invoking latexmk... done.

Could not find log file test.log!
Output from compilation:

Latexmk: This is Latexmk, John Collins, 1 January 2015, version: 4.41.
Latexmk: Changing directory to './'
Rule 'pdflatex': File changes, etc:
   Changed files, or newly in use since previous run(s):
      'test.tex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex  -interaction=nonstopmode -synctex=1 -recorder -output-directory="/home/carsten/git/rtsys/theses/csp-mt/thesis"  "test.tex"'
------------
Latexmk: applying rule 'pdflatex'...
This is pdfTeX, Version 3.14159265-2.6-1.40.16 (TeX Live 2015/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(/home/carsten/git/rtsys/theses/csp-mt/thesis/test.tex
LaTeX2e <2016/02/01>
Babel <3.9q> and hyphenation patterns for 10 language(s) loaded.
(/usr/share/texlive/texmf-dist/tex/latex/base/book.cls
Document Class: book 2014/09/29 v1.4h Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/bk10.clo))
(/home/carsten/git/rtsys/theses/csp-mt/thesis/test.aux) [1{/var/lib/texmf/fonts
/map/pdftex/updmap/pdftex.map}]
(/home/carsten/git/rtsys/theses/csp-mt/thesis/test.aux) )</usr/share/texlive/te
xmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb>
Output written on /home/carsten/git/rtsys/theses/csp-mt/thesis/test.pdf (1 page
, 10608 bytes).
SyncTeX written on /home/carsten/git/rtsys/theses/csp-mt/thesis/test.synctex.gz.
Transcript written on /home/carsten/git/rtsys/theses/csp-mt/thesis/test.log.
'test.fls' and '/home/carsten/git/rtsys/theses/csp-mt/thesis/test.fls' are identical (not copied) at /usr/bin/latexmk line 6657.
Latexmk: Log file says output to 'test.pdf'
Latexmk: Log file says output to 'test.pdf'
Latexmk: ===For rule 'pdflatex', actual output 'test.pdf'
       ======appears not to match expected output '/home/carsten/git/rtsys/theses/csp-mt/thesis/test.pdf'
Latexmk: Undoing directory change
Latexmk: All targets (/home/carsten/git/rtsys/theses/csp-mt/thesis/test.pdf) are up-to-date
```

**Comment by ig0774** [comment]
Thanks for the really detailed information!

I have a slight suspicion of what's going on. My guess is that you have another plugin with an `on_post_save` or `on_post_save_async` event that runs `os.chdir()` which interferes with LaTeXTools using `os.chdir()` to ensure we are in the right directory. Fortunately, this is easily fixable using the `makePDF.py` file [here](https://gist.githubusercontent.com/ig0774/c5101d84a4fa4af818092b09deb30c31/raw/923fce26dc733b337122fdfe853429a2c6ab867f/makePDF.py).

To use this, download the file, save it somewhere. From ST, launch the command palette (`C-shift-P`) and select **Preferences: Browse Packages**, find the LaTeXTools package and replace the `makePDF.py` file with the new version. You shouldn't need to restart. Try the build again and let me know if it works.

There may be one or two other things that need to change in other builders before this can be released, but it should solve your issue, if I'm right about what's happening.

**Comment by caspre24** [comment]
Wow, thanks for the quick solution! =) Your new `makePDF.py` fixes the problem!

To be sure (and just out of curiosity) I tried disabling all packages except LaTeXTools and as you suspected, the old `makePDF.py` worked as well.

One last question: Do I need to take any precautions for future updates regarding the modified `makePDF.py`?

**Comment by ig0774** [comment]
Thanks for taking the time to test! The fix for `makePDF.py` will be in the new release, so no need to worry.

**Comment by ig0774** [comment]
This fix has been released in v3.10.0.
